### ISSUE TITLE
Allow `tags` in `generated-index` sidebar item

### ISSUE THREAD

**Issue by Dr-Electron** [facebook/docusaurus#9841]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Description

As tags are more useful then keywords it would make sense to allow them too in the `generated-index` option

### Motivation

Tags are more useful but not supported. Keywords are. Would be nice to have at least both

### Self-service

- [X] I'd be willing to contribute this feature to Docusaurus myself.

**Comment by Josh-Cena** [comment]
The "keywords" are just SEO meta like "image" and "description" so they are very trivial to add (every single page supports these features). "Tags" are much more complex to build because they require treating these generated indices as actual doc pages.

**Comment by slorber** [comment]
Yes tags and keywords are totally unrelated. And adding tags to category generated index would not be so easy, for little ROI IMHO.

---

Considering you can link a category to an actual doc, and render a similar UI through the `DocCardList` component:
https://docusaurus.io/docs/sidebar/items#embedding-generated-index-in-doc-page

Wouldn't this be a good enough workaround to just link the category to a doc, and put tags on that doc? 

This would lead to a similar result IMHO.

**Comment by Dr-Electron** [comment]
I know that the changes are not trivial. I was just asking as a past discussion on the keyword lead to keywords basically being useless and so we replaced them everywhere to tags as that was the plan from the beginning anyway. We just used keyword in the beginning as we thought we could get both: indexing of pages and SEO. But as this was not the case we switched now.

But @slorber suggestion should work completely fine. I didn't think about it. Although for completeness you might want to think about supporting tags. For my use case this issue can be closed tho if you want. I will leave the decision up to you. Thanks for your awesome work ❤️
